include/openssl/core_names.h

```diff
@@ -94,6 +94,37 @@ extern "C" {
 #define OSSL_MAC_NAME_KMAC128       "KMAC128"
 #define OSSL_MAC_NAME_KMAC256       "KMAC256"
 
+/* KDF / PRF parameters */
+/* Note that "secret" and "key" are equivalent */
+#define OSSL_KDF_PARAM_SECRET       "secret"    /* octet string */
+#define OSSL_KDF_PARAM_KEY          "key"       /* octet string */
+#define OSSL_KDF_PARAM_SALT         "salt"      /* octet string */
+#define OSSL_KDF_PARAM_PASSWORD     "password"  /* octet string */
+/* Note that "md" and "digest" are equivalent */
+#define OSSL_KDF_PARAM_MD           "md"        /* utf8 string */
+#define OSSL_KDF_PARAM_DIGEST       "digest"    /* utf8 string */
+#define OSSL_KDF_PARAM_CIPHER       "cipher"    /* utf8 string */
+/* Note that "algorithm" can be used instead of "md", "digest" or "cipher" */
+#define OSSL_KDF_PARAM_ALGORITHM    "algorithm" /* utf8 string */
+#define OSSL_KDF_PARAM_MAC          "mac"       /* utf8 string */
+#define OSSL_KDF_PARAM_MAC_SIZE     "mac_size"  /* size_t */
+#define OSSL_KDF_PARAM_PROPERTIES   "properties" /* utf8 string */
+#define OSSL_KDF_PARAM_ITER         "iter"      /* unsigned int */
+#define OSSL_KDF_PARAM_MODE         "mode"      /* utf8 string */
+#define OSSL_KDF_PARAM_MODE_INT     "mode_int"  /* int */
+#define OSSL_KDF_PARAM_PKCS5        "pkcs5"     /* int */
+#define OSSL_KDF_PARAM_UKM          "ukm"       /* octet string */
+#define OSSL_KDF_PARAM_CEK_ALG      "cek_alg"   /* utf8 string */
+#define OSSL_KDF_PARAM_N            "n"         /* uint64_t */
+#define OSSL_KDF_PARAM_R            "r"         /* uint32_t */
+#define OSSL_KDF_PARAM_P            "p"         /* uint32_t */
```

Comment:
We probably need to discuss names. "n", "r" and "p" are specific to scrypt. That should at least be reflected in the macro name. Regarding the strings, we may have reason for debate, but have a look at the DH params below, that might suggest slightly different names.

Reply:
I renamed the defines but no the names themselves -- the old PKEY API uses _N_, _r_ and _p_ so easiest to stay with these (although I've used _n_ instead of _N_).